blockchain/tx_list.go

```diff
@@ -326,7 +327,10 @@ func (l *txList) Filter(senderBalance *big.Int, pool *TxPool) (types.Transaction
 			feePayer, _ := tx.FeePayer()
 			feePayerBalance := pool.getBalance(feePayer)
 			feeRatio, isRatioTx := tx.FeeRatio()
-			if isRatioTx {
+
+			if sender == feePayer {
```

Comment:
hmm? If it is a ratio tx, the comparison is right? I think this is to be included in the case that it is not ratio tx.

Reply:
i think checking sender==feePayer condition is necessary for both fee delegate valueTransfer & ~ with ratio.
whether or not the fee delegated Transaction is ratioTx, the Tx should be rejected if tx.Cost() > sender balance and vice versa.

Reply:
@JayChoi1736 What if the below case?
sender == feePayer && ratio == 10% ? With this code, it fallbacks the first condition. Is this your intention?

Reply:
@kjhman21 yes. 
when the ratio is 10% and sender == feePayer,
check logic from prior code is
senderBalance < (tx.value + tx.fee&ast;0.9) || senderBalance < (tx.fee&ast;0.1)
and changed logic is 
senderBalance < tx.Cost()

1) obviously if this statement is **TRUE**, changed logic will return **TRUE** because tx.Cost is always bigger than (tx.value + tx.fee&ast;0.9) , tx.fee&ast;0.1

2) but when the statement is **FALSE**, there are two cases. 
2-1) if senderBalance > tx.Cost(), changed logic will also return **FALSE**
2-2) if senderBalance < tx.Cost(), changed logic will return **TRUE** (and the filter will remove the Tx) 
   - but this one is intended because in case of senderBalance > (tx.value + tx.fee&ast;0.9) && senderBalance > (tx.fee&ast;0.1) && senderBalance < tx.Cost() , the Tx will not be executed and be pended in tx pool.

Please let me know if I've missed out anything.

Reply:
Oh, now I understood what problem you were trying to solve. Okay. So, this is kind of a shortcut before checking sender and fee payer balance check.

I rather suggest that do not add 'else' to reduce confusion in the next conditional statement.